helpers/cookieHelper.js

```diff
@@ -58,6 +52,12 @@ export const getUserCookie = (withParse = false) => {
   if (typeof window === 'undefined') {
     return false;
   }
+
+  const impersonateUser = getCookie('impersonateUser');
+  if (impersonateUser && withParse) {
+    return JSON.parse(impersonateUser);
+  }
```

Comment:
what if we have impersonateUser cookie but without parse?
you need another condition here